### PR TITLE
Show the folder path a file is in, in title of the window

### LINKED ISSUE #66660: Add a title variable to show the folder path a file is in
Issue Type: <b>Bug</b>

**Description:**
Recently, the "folderPath" entry ceased showing the entire folder path, instead, showing the path of the top-level open folder.

**To Reproduce:**
1. In _User Settings_ > _Window: Title_ > enter "${folderPath}"

Result: 
"/Users/Development/myProject"

Expected result:
"/Users/Development/myProject/myFolder"

**Example:**
_Window: Title_ value:
 `${dirty}${rootName}${separator}${activeEditorShort}${separator}${folderPath}`

Expected title:
`dev - ExecutionEnvironmentSection - C:\dev\eclipse.pde.ui\ui\org.eclipse.pde.ui\src\org\eclipse\pde\internal\ui\editor\plugin\ExecutionEnvironmentSection.java`

Title Produced:
`dev - ExecutionEnvironmentSection - C:\dev`

Example image: https://i.imgur.com/eZ1qJkt.png

VS Code version: Code 1.30.2 (61122f88f0bf01e2ac16bdb9e1bc4571755f5bd8, 2019-01-07T22:54:13.295Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: unavailable_software<br>checker_imaging: disabled_off<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: enabled_readback<br>webgl2: unavailable_off|
|Memory (System)|31.90GB (5.37GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
mssql|ms-|1.4.0
PowerShell|ms-|1.10.2
proto|pet|0.0.2
java|red|0.36.0
vscode-java-debug|vsc|0.16.0
vscode-java-dependency|vsc|0.3.0
vscode-java-pack|vsc|0.5.0
vscode-java-test|vsc|0.13.0
vscode-maven|vsc|0.13.0

</details>

### PR DESCRIPTION
**Pull request by vedipen**
Closes #66660
* Adds options in the _User Settings > Window: Title_ for showing the folder path a file is in.
* Simplified descriptions for variables

Added variables - 
* `${activeFolderShort}`: the name of the folder the file is contained in
* `${activeFolderMedium}`: the path of the folder the file is contained in, relative to the workspace folder
* `${activeFolderLong}`: the full path of the folder the file is contained in

Resulting view 
screenshot

### CODE REVIEW

File: src/vs/workbench/browser/parts/titlebar/titlebarPart.ts [hunk 1]
```diff
@@ -228,15 +228,18 @@ export class TitlebarPart extends Part implements ITitleService {
 	/**
 	 * Possible template values:
 	 *
-	 * {activeEditorLong}: e.g. /Users/Development/myProject/myFolder/myFile.txt
```
Comment (vedipen) [inline src/vs/workbench/browser/parts/titlebar/titlebarPart.ts]: redundant, making it in par with `{folderPath}` example

File: src/vs/workbench/browser/parts/titlebar/titlebarPart.ts [hunk 2]
```diff
@@ -259,6 +262,9 @@ export class TitlebarPart extends Part implements ITitleService {
 		const activeEditorShort = editor ? editor.getTitle(Verbosity.SHORT) : '';
 		const activeEditorMedium = editor ? editor.getTitle(Verbosity.MEDIUM) : activeEditorShort;
 		const activeEditorLong = editor ? editor.getTitle(Verbosity.LONG) : activeEditorMedium;
+		const activeFolderShort = editor ? (paths.dirname(editor.getTitle(Verbosity.LONG)) === '.' ? '' : paths.basename(paths.dirname(editor.getTitle(Verbosity.LONG)))) : '';
```
Comment (vedipen) [inline src/vs/workbench/browser/parts/titlebar/titlebarPart.ts]: additional check for tabs like _"Settings"_

**Review by bpasero — CHANGES_REQUESTED** [review]
@vedipen I am not a big fan of the logic in `titlebarPart.ts` to get the labels for short, medium and long folder. You implicitly assume that the label you get from the input is already a path, but that might not be true. A better solution imho would be to get the resource from the input and then do what the `fileEditorInput` does, e.g. here https://github.com/Microsoft/vscode/blob/cfd5d35/src/vs/workbench/parts/files/common/editors/fileEditorInput.ts#L171

**Comment by vedipen** [comment]
@bpasero  Thanks for the feedback. Updated the logic.

**Review by bpasero — CHANGES_REQUESTED** [review]
@vedipen I think we can still improve this a bit, specifically I think we do not need to check for `file` scheme but can try to support this for any resource. This would help in cases where the user opens remote resources which are not local.

As such I suggest:
* continue to use the `toResource(editor, { supportSideBySide: true })` but remove the check for `file` scheme
* use `resources.dirname()` instead of `paths.dirname()` which takes care of resources
* pass that into `this.labelService.getUriLabel` as before
* use `resources.basename` instead of `paths.basename` for the short label

**Comment by vedipen** [comment]
@bpasero Is this ideal or should we add a function for getting these paths (or maybe use [`getDescription`](https://github.com/Microsoft/vscode/blob/cfd5d358efa49c9dd3383bb8863f56a4c9997e5c/src/vs/workbench/parts/files/common/editors/fileEditorInput.ts#L146))?

**Comment by bpasero** [comment]
Thanks I think its fine. I will push small tweaks after to clean up a bit.

**Pull request merged**